pathology/multiple_instance_learning/panda_mil_train_evaluate_pytorch_gpu.py

```diff
@@ -0,0 +1,546 @@
+        data_list_key="validation",
+        base_dir=args.data_root,
+    )
+
+    if args.quick:  # for debugging on a small subset
+        training_list = training_list[:16]
+        validation_list = validation_list[:16]
+
+    train_transform = Compose(
+        [
+            LoadImageD(keys=["image"], reader=WSIReader, backend="TiffFile", dtype=np.uint8, level=1, image_only=True),
+            LabelEncodeIntegerGraded(keys=["label"], num_classes=args.num_classes),
+            TileOnGridd(
+                keys=["image"],
+                tile_count=args.tile_count,
+                tile_size=args.tile_size,
+                random_offset=True,
+                background_val=255,
+                return_list_of_dicts=True,
+            ),
+            RandFlipd(keys=["image"], spatial_axis=0, prob=0.5),
+            RandFlipd(keys=["image"], spatial_axis=1, prob=0.5),
+            RandRotate90d(keys=["image"], prob=0.5),
+            ScaleIntensityRangeD(keys=["image"], a_min=np.float32(255), a_max=np.float32(0)),
+            ToTensord(keys=["image", "label"]),
+        ]
+    )
+
+    valid_transform = Compose(
+        [
+            LoadImageD(keys=["image"], reader=WSIReader, backend="TiffFile", dtype=np.uint8, level=1, image_only=True),
+            LabelEncodeIntegerGraded(keys=["label"], num_classes=args.num_classes),
+            TileOnGridd(
+                keys=["image"],
+                tile_count=None,
+                tile_size=args.tile_size,
+                random_offset=False,
+                background_val=255,
+                return_list_of_dicts=True,
+            ),
+            ScaleIntensityRangeD(keys=["image"], a_min=np.float32(255), a_max=np.float32(0)),
+            ToTensord(keys=["image", "label"]),
+        ]
+    )
+
+    dataset_train = Dataset(data=training_list, transform=train_transform)
+    dataset_valid = Dataset(data=validation_list, transform=valid_transform)
+
+    train_sampler = DistributedSampler(dataset_train) if args.distributed else None
+    val_sampler = DistributedSampler(dataset_valid, shuffle=False) if args.distributed else None
+
+    train_loader = torch.utils.data.DataLoader(
+        dataset_train,
+        batch_size=args.batch_size,
+        shuffle=(train_sampler is None),
+        num_workers=args.workers,
+        pin_memory=True,
+        multiprocessing_context="spawn",
+        sampler=train_sampler,
+        collate_fn=list_data_collate,
+    )
+    valid_loader = torch.utils.data.DataLoader(
+        dataset_valid,
+        batch_size=1,
+        shuffle=False,
+        num_workers=args.workers,
+        pin_memory=True,
+        multiprocessing_context="spawn",
+        sampler=val_sampler,
+        collate_fn=list_data_collate,
+    )
+
+    if args.rank == 0:
+        print("Dataset training:", len(dataset_train), "validation:", len(dataset_valid))
+
+    # os.environ["TORCH_HOME"] = "../../torchhome" #TODO, remove later
```

Comment:
Could you please help remove according to the comment?

Thanks.

Reply:
ok